### DISCUSSION TITLE
Make rnodeconf available in android

### DISCUSSION THREAD

**Discussion by LaneaLucy** [markqvist/Reticulum#170] [General]
Now rnodeconf is integrated into rns, so it should be in the android app. If you now make it available from the GUI, you can update rnode's or init new one's. This would make it very easy to manage everything without need for a second device in a mobile scenario.
Thx for your work

**Comment by markqvist** [comment]
Yes, this would be very nice. There's probably a few minor additions that are needed to make it work under the different OS structure of Android, but all in all, it should be totally doable.

The biggest hurdle will probably be getting the esp32 flashing tool to work under Android, but even that is not something I think will be too much of an issue.

So yeah, it should definitely be possible, and I agree, would be very useful.

**Comment by LaneaLucy** [comment]
Shouldn't the flashing tool be base on c? If I remember right, you can include c code into android apps

**Comment by markqvist** [comment]
For some platforms it is, like AVR and ARM, but for ESP32 it is written in Python.

**Comment by LaneaLucy** [comment]
Shouldn't make it this then even easier? I mean at max someone need to replace the serial function in it

**Comment by LaneaLucy** [comment]
Found a c implementation that can easy being ported

https://github.com/espressif/esp-serial-flasher

**Comment by LaneaLucy** [comment]
And here is a java port

https://github.com/bdureau/ESPLoader

**Comment by LaneaLucy** [comment]
If I didn't hate java that much I would try to implement it myself and gave it to you, but I really, REALLY hate java

**Comment by markqvist** [comment]
> If I didn't hate java that much I would try to implement it myself and gave it to you, but I really, REALLY hate java

I feel your pain there... I am not a particular fan either.

**Comment by LaneaLucy** [comment]
So what do you think about the resources I found? How much work would it be with them? And is there anything I can motivate you to implement it?

**Reply by markqvist** [reply]
It's not really that the motivation is lacking, more that I am trying to balance five or six major software projects already, and also have to somehow pay the bills ;) Maybe I really should start trying to get some broader community and fund support going.

It's pretty high on my personal list of priorities, but I can't really make any promises currently as to when exactly it will be ready.

**Reply by markqvist** [reply]
The current step is to figure out a really efficient way to include RNS in Android apps. The kivy/buildozer build system that I used for Sideband is really convenient, but it is not that flexible when it comes down to it. I need to find a good way to compile everything as a shared library that can be loaded dynamically by any app, or run as a background service. That's what I am currently trying to find the optimal solution for.

**Reply by LaneaLucy** [reply]
Then why not make rns a dependency standalones app?

**Reply by markqvist** [reply]
Still need the above to do that ;)

**Reply by LaneaLucy** [reply]
I have good experience with termux, but maybe this is a bit to difficult for some people

**Reply by markqvist** [reply]
Yes, for some definitely, but most technical people will like it that way, it is also what I'd prefer myself.

It's coming, hopefully sooner rather than later :)
